src/services/lease-service/routes/applicants.ts

```diff
@@ -118,142 +118,109 @@ export const routes = (router: KoaRouter) => {
+          ctx.status = 403
           ctx.body = {
-            reason:
-              'Applicant is not a current or coming tenant in the property',
+            reason: 'User is not a current or coming tenant in the property',
           }
-          ctx.status = 403
           return
         }
 
-        //if applicant has no parking space contracts but is a tenant in the property
-        if (
-          !detailedApplicant.parkingSpaceContracts ||
-          detailedApplicant.parkingSpaceContracts.length == 0
-        ) {
-          //applicant is eligible for parking space, applicationType for application should be 'additonal'
+        //if subject has no parking space contracts but is a tenant in the property
+        if (!contact.data.parkingSpaceContracts?.length) {
+          //subject is eligible for parking space, applicationType for application should be 'additional'
+          //todo: clients needs an easy way to act upon success. Pass the needed applicationType in response?
+          ctx.status = 200 // accepted
           ctx.body = {
             reason:
-              'User does not have any active parking space contracts in the listings residential area',
+              'User is a tenant in the property and does not have any active parking space contracts in the listings residential area. User is eligible to apply with applicationType additional.',
           }
-          ctx.status = 403
           return
         }
 
-        //1. fetch the estateCode for each of the applicant's parking space contracts
-        //validation:
-        //2. Check if any of the users parking space contracts matches the listings estate code
-        //3. if any parking space contract matches the listing estatecode, user needs to replace that parking space contract
-        //4. else, the user can apply with applicationType 'additional'
-
-        //todo: refactor and move to property rules validator?
-        let applicantNeedsToReplaceContractToBeAbleToApply = false
-        for (const parkingSpaceContract of detailedApplicant.parkingSpaceContracts) {
-          //
-          const parkingSpaceEstateCode =
-            await getEstateCodeFromXpandByRentalObjectCode(
-              parkingSpaceContract.rentalPropertyId
-            )
-
-          if (parkingSpaceEstateCode != undefined) {
-            if (listingEstateCode == parkingSpaceEstateCode) {
-              applicantNeedsToReplaceContractToBeAbleToApply = true
-              break
-            }
-          }
-        }
+        const getParkingSpacePropertyInfo =
+          contact.data.parkingSpaceContracts.map((lease) =>
+            getEstateCodeFromXpandByRentalObjectCode(lease.rentalPropertyId)
+          )
+
+        const parkingSpaceEstateCodes = await Promise.all(
+          getParkingSpacePropertyInfo
+        ).then((res) => res.filter(Boolean).map((r) => r?.estateCode))
+
+        const subjectNeedsToReplaceParkingSpace = parkingSpaceEstateCodes.some(
```

Comment:
I did some testing and added a few comments previously. I think we need to make it easy for clients to act upon the response and scrap the idea to only use status codes to decide the action for the client. 

A proposal is to always return 200 if the validation passes and the allowed application type (additional, replace) in the body. What do you think?

Reply:
Sounds good to me!

Reply:
I saw that you made one commit. Unfortunately I lost it during the very cumbersome conflict merging process.. 
You probably have it on your local environment. When you get back maybe we can rebase it on top of my changes?